tests/store/artifact/utils/test_model_utils.py

```diff
@@ -51,19 +53,46 @@ def test_parse_models_uri_with_stage(uri, expected_name, expected_stage):
     ],
 )
 def test_parse_models_uri_with_latest(uri, expected_name):
-    (name, version, stage) = _parse_model_uri(uri)
+    (name, version, stage, alias) = _parse_model_uri(uri)
     assert name == expected_name
     assert version is None
     assert stage is None
+    assert alias is None
+
+
+@pytest.mark.parametrize(
+    ("uri", "expected_name", "expected_alias"),
+    [
+        ("models:/AdsModel1@Champion", "AdsModel1", "Champion"),
+        ("models:/AdsModel1@champion", "AdsModel1", "champion"),  # case insensitive
+        ("models:/AdsModel1@cHAmpion", "AdsModel1", "cHAmpion"),  # case insensitive
+        ("models:/Ads Model 1@challenger", "Ads Model 1", "challenger"),
+        ("models://scope:key/Ads Model 1@None", "Ads Model 1", "None"),
+    ],
+)
+def test_parse_models_uri_with_alias(uri, expected_name, expected_alias):
+    (name, version, stage, alias) = _parse_model_uri(uri)
+    assert name == expected_name
+    assert version is None
+    assert stage is None
+    assert alias == expected_alias
 
 
 @pytest.mark.parametrize(
     "uri",
     [
         "notmodels:/NameOfModel/12345",  # wrong scheme with version
         "notmodels:/NameOfModel/StageName",  # wrong scheme with stage
+        "notmodels:/NameOfModel@alias",  # wrong scheme with alias
         "models:/",  # no model name
+        "models:/ /Stage",  # empty name
+        "models:/Name",  # no specifiers
+        "models:/Name/",  # empty suffix
+        "models:/Name@",  # empty alias
         "models:/Name/Stage/0",  # too many specifiers
+        "models:/Name/Stage@Alias",  # stage and alias both specified
+        "models:/Name@alias/Stage",  # Stage and alias both specified
+        "models:/Name@Alias@other",  # too many aliases
```

Comment:
Minor: can we test for some other invalid characters in the alias name? e.g. some UTF8 characters? The regex looks fine to me though